### PR TITLE
Update for Dataporten closing #3580

### LINKED ISSUE #3580: Custom OAuth not validating.
Your Rocket.Chat version: latest from docker

When adding a custom OAuth provider (https://docs.dataporten.no/docs/authorization/), I get this as an error in the meteor console: 

``` Exception while invoking method 'login' Error: Service data for service testapp must include id```

I have my suspicions about the returned array is in a different form wanted by Rocket.Chat, so is there any way of specifying this, creating a plugin to take this in mind?

Here's the format the data is returned 

```
{
    "user": {
        "userid": "76a7a061-3c55-430d-8ee0-6f82ec42511f",
        "userid_sec": ["feide:andreas@uninett.no"],
        "name": "Full Display name",
        "email": "email@address.something",
        "profilephoto": "photo:ID"
    },
    "audience": "e8160a77-58f8-4006-8ee5-ab64d17a5b1e"
}
```

More info here: https://docs.dataporten.no/docs/oauth-authentication/

### PR DESCRIPTION
**Pull request by kasperrt**
@RocketChat/core 

Closes #3580

### CODE REVIEW

File: packages/rocketchat-custom-oauth/custom_oauth_server.coffee [hunk 1]
```diff
@@ -122,7 +122,16 @@ class CustomOAuth
...
 				identity.id = identity.CharacterID
-
+			
+			# Fix Dataporten having 'user.userid' instead of 'id'
+			if identity?.user.userid and not identity.id
+				identity.id = identity.user.userid
+				identity.email = identity.user.email
+			
+			# Fix general 'userid' instead of 'id' from provider
+			if identity?.userid and not identityid
```
Comment (rodrigok) [inline packages/rocketchat-custom-oauth/custom_oauth_server.coffee]: Shouldn't `identityid` be `identity.id`?

File: packages/rocketchat-custom-oauth/custom_oauth_server.coffee [hunk 2]
```diff
@@ -135,7 +144,7 @@ class CustomOAuth
 				serviceData: serviceData
 				options:
 					profile:
-						name: identity.name or identity.username or identity.nickname or identity.CharacterName
+						name: identity.name or identity.username or identity.nickname or identity.CharacterName or identity.user.name
```
Comment (rodrigok) [inline packages/rocketchat-custom-oauth/custom_oauth_server.coffee]: Can you protect the code replacing `identity.user.name` by `identity.user?.name`

**Comment by rodrigok** [comment]
Thanks by the contribution 😃

File: packages/rocketchat-custom-oauth/custom_oauth_server.coffee (hunk 2 above)
Comment (kasperrt) [inline packages/rocketchat-custom-oauth/custom_oauth_server.coffee]: ``identity.user?.name``` or ```identity.user?.name?```

Comment (rodrigok) [inline packages/rocketchat-custom-oauth/custom_oauth_server.coffee]: identity.user?.name

Comment (kasperrt) [inline packages/rocketchat-custom-oauth/custom_oauth_server.coffee]: Done! :)

**Comment by rodrigok** [comment]
LGTM

**Comment by rodrigok** [comment]
@RocketChat/core I need one more review here 😸

**Pull request merged**